homeassistant/components/motionmount/__init__.py

```diff
@@ -0,0 +1,69 @@
+"""The Vogel's MotionMount integration."""
+from __future__ import annotations
+
+import socket
+
+import motionmount
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT, Platform
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.device_registry import format_mac
+
+from .const import DOMAIN, EMPTY_MAC
+from .coordinator import MotionMountCoordinator
+
+PLATFORMS: list[Platform] = [
+    Platform.NUMBER,
+]
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Vogel's MotionMount from a config entry."""
+
+    hass.data.setdefault(DOMAIN, {})
```

Comment:
We usually don't store data in `hass.data` until after we are past the point we know we are going to proceed with setup. Ie `async_config_entry_first_refresh` has not failed